code/modules/hippie/sound/sound.dm

```diff
@@ -3,148 +3,215 @@
-							pressure_factor = max((pressure - SOUND_MINIMUM_PRESSURE)/(ONE_ATMOSPHERE - SOUND_MINIMUM_PRESSURE), 0)
-					else //space
-						pressure_factor = 0
-
-					var/distance = get_dist(mob_loc, source_turf)
-
-					if(distance <= 1)
-						pressure_factor = max(pressure_factor, 0.15) //touching the source of the sound
-
-					S.volume *= pressure_factor
-
-					if(isliving(M))
-						var/mob/living/L = M
-						if (L.hallucination)
-							S.environment = SOUND_ENVIRONMENT_PSYCHOTIC
-						else if (L.druggy)
-							S.environment = SOUND_ENVIRONMENT_DRUGGED
-						else if (L.drowsyness)
-							S.environment = SOUND_ENVIRONMENT_DIZZY
-						else if (L.confused)
-							S.environment = SOUND_ENVIRONMENT_DIZZY
-						else if (L.sleeping)
-							S.environment = SOUND_ENVIRONMENT_UNDERWATER
-
-					if(S.volume <= 0)
-						return
-
-				S.x = source.x - mob_loc.x
-				S.z = source.y - mob_loc.y
+				//if(get_dist(M, source) >= 30) return // hard attentuation i guess
+				S.x = source.x - Mloc.x
+				S.z = source.y - Mloc.y //Since sound coordinates are 3D, z for sound falls on y for the map.  BYOND.
 				S.y = 0
+				S.volume *= attenuate_for_location(Mloc)
 				M << S
 				S.volume = vol
 
-/atom/proc/playsound_local(turf/turf_source, soundin, vol as num, vary, frequency, falloff, surround = 1)
-
-	var/sound/S = generate_sound(turf_source, soundin, vol, vary, 0, frequency)
-	S.volume = vol
-
-	if (vary)
-		S.frequency = rand(725, 1250) / 1000 * frequency
-	else
-		S.frequency = frequency
-
-	if(isturf(turf_source))
-		var/turf/T = get_turf(src)
-
-		var/pressure_factor = 1
-		var/datum/gas_mixture/hearer_env = T.return_air()
-		var/datum/gas_mixture/source_env = turf_source.return_air()
+		//pool(S)
+
+/atom/proc/playsound_local(var/atom/source, soundin, vol as num, vary, extrarange as num, pitch = 1)
+
+	switch(soundin)
+		if ("shatter") soundin = pick(sounds_shatter)
+		if ("explosion") soundin = pick(sounds_explosion)
+		if ("sparks") soundin = pick(sounds_sparks)
+		if ("rustle") soundin = pick(sounds_rustle)
+		if ("punch") soundin = pick(sounds_punch)
+		if ("clownstep") soundin = pick(sounds_clownstep)
+		if ("swing_hit") soundin = pick(sounds_hit)
+		if ("hiss") soundin = pick(sounds_hiss)
+		if ("pageturn") soundin = pick(sounds_pageturn)
+		if ("bodyfall") soundin = pick(sounds_bodyfall)
+		if ("gunshot") soundin = pick(sounds_gunshot)
+		if ("ricochet") soundin = pick(sounds_ricochet)
+		if ("terminal_type") soundin = pick(sounds_terminal)
 
-		if(hearer_env && source_env)
-			var/pressure = min(hearer_env.return_pressure(), source_env.return_pressure())
-			if(pressure < ONE_ATMOSPHERE)
-				pressure_factor = max((pressure - SOUND_MINIMUM_PRESSURE)/(ONE_ATMOSPHERE - SOUND_MINIMUM_PRESSURE), 0)
-		else
-			pressure_factor = 0
+	if(islist(soundin))
+		soundin = pick(soundin)
 
-		var/distance = get_dist(T, turf_source)
-		if(distance <= 1)
-			pressure_factor = max(pressure_factor, 0.15)
+	var/sound/S
```

Comment:
This code is duplicated later in this file, can you move this into a generic proc that returns a sound?